### PR TITLE
DCOS_OSS-4116: Support Universe packages with minimal schema

### PR DESCRIPTION
**Pull request by nLight**
This PR adds support of packages with minimal schema `{type: "object"}`. 

### Preparation

To test this we need a bad package. I created such. To use to we need to add a universe repo containing it to dcos.

Hard way: https://github.com/mesosphere/universe :feelsgood: 
Easy way: down below :)

#### 1. Launch the following app on your cluster:
```
{
  "id": "/universe",
  "constraints": [
    [
      "hostname",
      "UNIQUE"
    ]
  ],
  "container": {
    "portMappings": [
      {
        "containerPort": 80,
        "hostPort": 8085,
        "protocol": "tcp",
        "servicePort": 10004
      }
    ],
    "type": "DOCKER",
    "volumes": [],
    "docker": {
      "image": "rozhkovd/universe:bug",
      "forcePullImage": true,
      "privileged": false,
      "parameters": []
    }
  },
  "cpus": 0.25,
  "disk": 0,
  "healthChecks": [
    {
      "gracePeriodSeconds": 120,
      "ignoreHttp1xx": false,
      "intervalSeconds": 30,
      "maxConsecutiveFailures": 3,
      "portIndex": 0,
      "timeoutSeconds": 5,
      "delaySeconds": 15,
      "protocol": "HTTP",
      "path": "/repo-empty-v3.json",
      "ipProtocol": "IPv4"
    }
  ],
  "instances": 1,
  "maxLaunchDelaySeconds": 3600,
  "mem": 128,
  "gpus": 0,
  "networks": [
    {
      "mode": "container/bridge"
    }
  ],
  "requirePorts": false,
  "fetch": []
}
```
#### 2. Add the repo

1. Settings -> Package Repositories -> +
2. Name: anything
3. URL: `http://universe.marathon.mesos:8085/repo`
4. Priority: 0

#### Valid package

1. Catalog
2. Pick any package like kafka/nginx/cassandra etc.
3. Review and run
4. The form should open with no errors

#### Invalid Package

1. Catalog
2. Pick `dcos_oss-616`
3. Review and run
4. 
screenshot 2018-09-14 at 15 03 16

## Trade-offs

1. Updated the underlying lib to get around the shallow schema bug
2. Extracted `getFirstTabAndField` to test in isolation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,7 +66,7 @@
     "react-dom": "15.4.1",
     "react-gemini-scrollbar": "2.1.5",
     "react-intl": "2.2.3",
-    "react-jsonschema-form": "0.50.0",
+    "react-jsonschema-form": "1.0.4",
```
Comment (jaxesn) [inline package.json]: Any breaking changes to be concerned about in this upgrade?

File: src/js/utils/FrameworkConfigurationUtil.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import Util from "#SRC/js/utils/Util";
+
+// eslint-disable-next-line import/prefer-default-export
+export function getFirstTabAndField(packageDetails) {
+  window.packageDetails = packageDetails;
```
Comment (jaxesn) [inline src/js/utils/FrameworkConfigurationUtil.js]: Why the global?

Comment (nLight) [reply src/js/utils/FrameworkConfigurationUtil.js]: 🙈

File: package.json (hunk 1 above)
Comment (nLight) [reply package.json]: There was one, it doesn't look it affects us at all.

**Pull request merged**